blas/impl/KokkosBlas1_axpby_unification_attempt_traits.hpp

```diff
@@ -0,0 +1,955 @@
+    // ******************************************************************
+    // Check 1/6: General checks
+    // ******************************************************************
+    static_assert(
+        Kokkos::is_execution_space_v<tExecSpace>,
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": tExecSpace must be a valid Kokkos execution space.");
+
+    static_assert(
+        (xyRank1Case && !xyRank2Case) || (!xyRank1Case && xyRank2Case),
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": one must have either both X and Y as rank 1, or both X and Y as "
+        "rank 2");
+
+    if constexpr (Kokkos::ArithTraits<
+                      AtInputScalarTypeY_nonConst>::is_complex == false) {
+      static_assert(
+          (Kokkos::ArithTraits<AtInputScalarTypeA_nonConst>::is_complex ==
+           false) &&
+              (Kokkos::ArithTraits<AtInputScalarTypeX_nonConst>::is_complex ==
+               false) &&
+              (Kokkos::ArithTraits<AtInputScalarTypeB_nonConst>::is_complex ==
+               false),
+          "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+          ": if Y is not complex, then A, X and B cannot be complex");
+    }
+
+    // ******************************************************************
+    // Check 2/6: YMV is valid
+    // ******************************************************************
+    static_assert(
+        Kokkos::is_view<YMV>::value,
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": Y is not a Kokkos::View.");
+    static_assert(
+        std::is_same<typename YMV::value_type,
+                     typename YMV::non_const_value_type>::value,
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": Y is const.  It must be nonconst, "
+        "because it is an output argument "
+        "(we must be able to write to its entries).");
+    static_assert(
+        Kokkos::SpaceAccessibility<tExecSpace,
+                                   typename YMV::memory_space>::accessible,
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": XMV must be accessible from tExecSpace");
+
+    // ******************************************************************
+    // Check 3/6: XMV is valid
+    // ******************************************************************
+    static_assert(
+        Kokkos::is_view<XMV>::value,
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": X is not a Kokkos::View.");
+    static_assert(
+        Kokkos::SpaceAccessibility<tExecSpace,
+                                   typename XMV::memory_space>::accessible,
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": XMV must be accessible from tExecSpace");
+
+    if constexpr (xyRank1Case) {
+      if (X.extent(0) == Y.extent(0)) {
```

Comment:
Check `if(X.extent(0) != Y.extent(0)) { ... }` instead of having an empty if clause.

Reply:
Done. Here I agree, because the understanding is pretty immediate in both cases (== or !=).